controller/array_action/array_mediator_interface.py

```diff
@@ -243,13 +230,13 @@ def get_object_by_id(self, object_id, object_type):
         raise NotImplementedError
 
     @abstractmethod
-    def create_snapshot(self, name, volume_name, pool_id=None):
+    def create_snapshot(self, volume_id, snapshot_name, pool=None):
         """
         This function should create a snapshot from volume in the storage system.
         Args:
-            name           : name of the snapshot to be created in the storage system
-            volume_name    : name of the volume to be created from
-            pool_id: pool_id of the volume to find the snapshot more efficiently.
+            volume_id : id of the volume to be created from
+            snapshot_name : name of the snapshot to be created in the storage system
+            pool : pool to create the snapshot in (and find it if already exist).
```

Comment:
I'm not sure

> (and find it if already exist)

is part of this method's responsibilities

Reply:
done

Reply:
I don't see any change

Reply:
wired.. done now

Reply:
should we have `(if not given, pool taken from source volume)` here as well?

Reply:
sure